### PR TITLE
feat: add UiPhoneNumber component

### PR DESCRIPTION
**Pull request by AlanLes**
# Related issue
Closes issue #9 

## Scope of work
- create the UiPhoneNumber component,
- create stories.ts and spec.ts files,
- write stories
- write unit tests

out2

### CODE REVIEW

**Comment by pspaczek** [comment]
@AlanLes is still WiP? :) If it's ready to CR please remove WIP from title <3

**Comment by AlanLes** [comment]
Yes @pspaczek it's still WIP - I'm coding it in some spare time, after hours, on weekends mostly :) ATM there are missing **stories** and **unit tests**, but if you think we can distinguish these into other issues/tasks, then it would be ready for review :)

**Comment by AlanLes** [comment]
@pspaczek it's ready to review 😄

**Review by ArkadiuszKrzymienski — CHANGES_REQUESTED** [review]
I left some comments 😃 Overall awesome job 💪 thank you @AlanLes 🤝

File: src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue [hunk 1]
```diff
@@ -0,0 +1,384 @@
+<template>
+  <section
+    v-if="!isLoading"
```
Comment (pspaczek) [inline src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue:3]: R: IMO it shouldn't be here. To be honest this component should be split into the smallest parts.
IMO wee need split it to 
* country selector to get phone codes and it should handle state where we waiting for codes 
* phone number input

**Review by pspaczek — CHANGES_REQUESTED** [review]

File: src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    class="ui-phone-number"
+  >
+    <fieldset class="ui-phone-number__fields">
+      <slot
+        name="legend"
+        v-bind="{ label: translation.phoneNumberLabel }"
+      >
+        <UiText
```
Comment (pspaczek) [inline src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue:11]: Q: why?

File: src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue [hunk 3]
```diff
@@ -0,0 +1,384 @@
+<template>
+  <section
+    v-if="!isLoading"
+    class="ui-phone-number"
+  >
+    <fieldset class="ui-phone-number__fields">
+      <slot
+        name="legend"
+        v-bind="{ label: translation.phoneNumberLabel }"
```
Comment (pspaczek) [inline src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue:9]:
```suggestion
        v-bind="{ translation }"
```

**Comment by pspaczek** [comment]
Sorry but this component needs a lot of love.
1. split it so smallest components
2. remove this v-if from root element. Why you want hide everything where you waiting for data for country codes :( 
3. Where we have figma for this component can you share it?

File: src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue [hunk 4]
```diff
@@ -0,0 +1,384 @@
+<template>
+  <section
```
[the remark this answers is not in the document]
Comment (AlanLes) [reply src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue:2]: Why not? Should I use a div instead? The `<section>` tag means that the content inside relates to a single theme, and the `<div>` tag is used as a block part of the webpage, and it doesn't convey any particular meaning. :) Obviously, I can change it to `<div>` if you want, of course :)

File: src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue (hunk 3 above)
Comment (AlanLes) [reply src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue:9]: Q: Why bind the whole object if in this `<slot>`'s scope, the only property used is `label`? 🤔 :D

**Comment by AlanLes** [comment]
> Sorry but this component needs a lot of love.
> 
> 1. split it so smallest components
> 2. remove this v-if from root element. Why you want hide everything where you waiting for data for country codes :(
> 3. Where we have figma for this component can you share it?

1. Okay, thank you! I'll do that! :D 
2. Yep, I'll do that as well :) 
3. https://www.figma.com/file/wz68QdkR6ldjmVrPYrazsr/Core-Symptom-Checker-UI-Library?node-id=7490%3A228243&t=lIRJ5sqGkyiDJO7K-4 :)

File: src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue (hunk 2 above)
Comment (AlanLes) [reply src/components/organisms/UiPhoneNumber/UiPhoneNumber.vue:11]: Why the legend? Why I used the UiText component? Or why what? Could you be more specific? :D

**Pull request closed without merging**